### DISCUSSION TITLE
ClientException

### DISCUSSION THREAD

**Discussion by Dan-Crane** [hawkkiller/sizzle_starter#323] [General]
Hi! Thanks for the cool template, i found something useful for me 🤗

Why [`ClientException`](https://github.com/hawkkiller/sizzle_starter/blob/c51ae3da7365247e4eeb03e4a63227f317ce80fc/lib/src/core/components/rest_client/src/exception/network_exception.dart#L40) extends `RestClientException`? Isn't it more correct to extend [`RestClientExceptionWithCause`](https://github.com/hawkkiller/sizzle_starter/blob/c51ae3da7365247e4eeb03e4a63227f317ce80fc/lib/src/core/components/rest_client/src/exception/network_exception.dart#L23)?

`RestClientExceptionWithCause` and `ClientException` are identical by the set of fields.

**Comment by hawkkiller** [comment]
You are correct! This will be fixed (if not already fixed in a new PR I'm slowly working on 😁)

**Reply by Dan-Crane** [reply]
I noticed that there is an exception called  [`InternalServerException`](https://github.com/hawkkiller/sizzle_starter/blob/adf99cc1906c7f033606693358d99564dcfce3eb/lib/src/core/components/rest_client/src/exception/network_exception.dart#L123), but it is not used anywhere. Maybe we should add a check for statusCode >= 500 inside the [`sendRequest`](https://github.com/hawkkiller/sizzle_starter/blob/adf99cc1906c7f033606693358d99564dcfce3eb/lib/src/core/components/rest_client/src/rest_client_dio.dart#L17) function, and if true, throw the `InternalServerException`?
